ledger/complete/wal/checkpoint_v6_reader.go

```diff
@@ -365,10 +371,12 @@ func getNodeByIndex(subtrieNodes [][]*node.Node, topLevelNodes []*node.Node, ind
 		return nil, fmt.Errorf("top trie [0] %v isn't nil", topLevelNodes[0])
 	}
 
-	if offset >= uint64(len(topLevelNodes)-1) {
-		return nil, fmt.Errorf("can not find node by index: %v in subtrieNodes %v, topLevelNodes %v", index, subtrieNodes, topLevelNodes)
+	nodePos := index - totalSubTrieNodeCount
+
+	if nodePos >= uint64(len(topLevelNodes)) {
+		return nil, fmt.Errorf("can not find node by index %v, nodePos >= len(topLevelNodes) => (%v > %v)",
+			index, nodePos, len(topLevelNodes))
 	}
 
-	// +1 because first item is always nil
-	return topLevelNodes[offset+1], nil
```

Comment:
Addressing comment from @fxamacker , I extracted a `getTopNodeByIndex` function

>Maybe this part of the function can be moved to the caller.

>This function can focus on getSubTrieNodeByIndex and the caller can get node like this:

>if node index == 0, node is nil
if node index <= totalSubTrieNodeCount, call this function to look up node in the subtrie groups
else, look up node in topLevelNodes after adjusting index